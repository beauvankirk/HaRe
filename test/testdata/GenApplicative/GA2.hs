module Main where
import Text.ParserCombinators.Parsec

parseStr :: CharParser () String 
parseStr = char '"' *> (many1 (noneOf "\"")) <* char '"'

objEntry :: CharParser () (String, String)
objEntry = do
  spaces
  str <- parseStr
  spaces
  char ':'
  spaces
  i <- many1 digit
  spaces
  return (str, i)

main = putStrLn "hello"
