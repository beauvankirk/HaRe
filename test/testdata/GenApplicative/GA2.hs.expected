module Main where
import Text.ParserCombinators.Parsec

parseStr :: CharParser () String
parseStr = char '"' *> (many1 (noneOf "\"")) <* char '"'

objEntry :: CharParser () (String, String)
objEntry = (,) <$> (spaces *> parseStr <* spaces <* char ':') <*> (spaces *> (many1 digit) <* spaces)

main = putStrLn "hello"
